### PR TITLE
Chunk oversized JSON string writes in JsonToBinaryStream

### PR DESCRIPTION
**Pull request by Amemoyoi**
This change avoids oversized raw writes in the proto3 JSON-to-binary streaming path.

  Changes:
  - chunk oversized writes in `ParseProto3Type::SetString()` instead of passing a `size_t` length directly into `WriteRaw(const void*, int)`
  - add a focused regression test for the chunking behavior at the `INT_MAX + 1` boundary

### CODE REVIEW

**Comment by esrauchg** [comment]
Thank you for raising this!

Serializing strings above 2 GiB is not supported by protobuf in general (our length-prefix format has an `int32` for sizes, so strings that large aren't representable in our main format at all).

I think we likely would prefer to serialize-fail if such a condition is hit rather than trying to gracefully chunk up the writes: do you think you could update the PR to do?

**Comment by Amemoyoi** [comment]
Thanks, that makes sense.
I’ve updated the PR to fail cleanly when a JSON string/bytes value exceeds the supported INT32_MAX serialization limit, instead of chunking the writes.

File: src/google/protobuf/json/internal/parser_traits.h [hunk 1]
```diff
@@ -200,13 +209,14 @@ struct ParseProto2Descriptor : Proto2Descriptor {
     }
   }
 
-  static void SetString(Field f, Msg& msg, absl::string_view x) {
+  static absl::Status SetString(Field f, Msg& msg, absl::string_view x) {
     RecordAsSeen(f, msg);
```
Comment (esrauchg) [inline src/google/protobuf/json/internal/parser_traits.h:213]: I think adding the CheckSuppportedJsonStringSize() at this point also makes sense; even though this path 'works', IMO better to have behavior parity with the other path, the hard limits in the binary format means that we won't very gracefully handle such cases downstream from here.

**Review by esrauchg — COMMENTED** [review]
Thank you!

File: src/google/protobuf/json/internal/parser_traits.h (hunk 1 above)
Comment (Amemoyoi) [reply src/google/protobuf/json/internal/parser_traits.h:214]: Makes sense, thanks. I’ve updated the reflective path as well so the size check is applied consistently across both paths.

**Review by esrauchg — APPROVED** [review]
Thanks!

**Comment by Amemoyoi** [comment]
Thanks again for the review.
I’ve synced the branch with main, and the blocking checks appear to be passing now.
Please let me know if there’s anything else needed from my side before merge.

**Review by esrauchg — APPROVED** [review]

**Comment by esrauchg** [comment]
Unfortunately looks like theres a build breakage with latest proposed change, are you able to fix?

Thanks!

**Comment by Amemoyoi** [comment]
Thanks for the heads up. The breakage was from defining the helper in the header with external linkage. I’ve updated it to be inline and pushed the fix.

**Comment by esrauchg** [comment]
Unfortunately looks like the json_test build error is still there on latest snapshot.

**Comment by Amemoyoi** [comment]
I updated the PR to reject oversized JSON string/bytes values instead of chunking them, matching the requested fail-
  fast behavior. I also added the missing direct Bazel dependency for the internal parser helper used by `json_test`
  and added explicit `<cstddef>` includes.

  The latest pushed commit is `caf7d4a9a`. Could you please re-run/enable the relevant CI when you have a chance?

**Review by esrauchg — APPROVED** [review]

**Comment by rgoldfinger6** [comment]
Please fix failing tests.

**Comment by tonyliaoss** [comment]
@Amemoyoi It seems like there are merge conflicts with this PR. Could you rebase to head?
